R/LSTsampleVariability.R

```diff
@@ -16,279 +16,288 @@
-    xBreaks <- jaspGraphs::getPrettyAxisBreaks(c(min - range / 4, max + range / 4))
-    xLimits <- range(xBreaks)
-    pdPlotObject <-  ggplot2::ggplot(data.frame(x = xBreaks), ggplot2::aes(x = x)) +
-      ggplot2::stat_function(fun = dunif, n = 100, args = list(min = min, max = max), geom = "area", fill = fillColor) +
-      ggplot2::stat_function(fun = dunif, n = 100, args = list(min = min, max = max))
-    if (returnData)
-      df <- data.frame(x = runif(n = 100000, min = min, max = max))
-  } else if (distribution == "skewed") {
-    if (skewDirection == "left")
-      skew <- skew * -1
-    xBreaks <- jaspGraphs::getPrettyAxisBreaks(c(mean - 3*sd, mean + 3*sd ))
-    xLimits <- range(xBreaks)
-    df <- data.frame(x = .scaledSkewedNormal(100000, xi = mean, omega = sd, alpha = skew))
-    pdPlotObject <-  ggplot2::ggplot(df, ggplot2::aes(x = x)) + ggplot2::geom_density(mapping = ggplot2::aes(y = ..density..),
-                                                                                      n = 2^7, bw = sd/3, fill = fillColor)
-  }
-  yBreaks <- jaspGraphs::getPrettyAxisBreaks(ggplot2::ggplot_build(pdPlotObject)$data[[1]]$y)
-  yLimits <- range(yBreaks)
-  if (showMean) {
-    yPos <- max(ggplot2::ggplot_build(pdPlotObject)$data[[1]]$y) * 1.1
-    pdPlotObject <- pdPlotObject + ggplot2::geom_vline(xintercept = mean, color = "darkmagenta", size = 1) + 
-      ggplot2::geom_label(data = data.frame(x = mean, y = yPos, label = gettextf("Mean: %.2f", mean)), 
-                          mapping = ggplot2::aes(x = x, y = y, label = label), color = "darkmagenta", size = 6)
-    yLimits <- range(c(yLimits, yPos))
-  }
-  pdPlotObject <- pdPlotObject + ggplot2::scale_x_continuous(breaks = xBreaks, limits = xLimits, name = "") +
-    ggplot2::scale_y_continuous(breaks = yBreaks, limits = yLimits, name = "Density") + 
-    jaspGraphs::themeJaspRaw() + jaspGraphs::geom_rangeframe()
+.arrangePlotMat <- function(plotList, col = 2, tbc = ""){
+  nPlots <- length(plotList)
+  plotMatDetails <- .getPlotMatDetails(nPlots, col, tbc)
+  rows <- plotMatDetails$rows
+  tbcIndex <- plotMatDetails$tbcIndex
+  emptyIndex <- plotMatDetails$emptyIndex
+  plotMat <- matrix(list(), rows, col)
+  index <- 0
   
-  if (!returnData){
-    return(pdPlotObject)
-  } else {
-    return(list(plotobject = pdPlotObject, data = df))
+  for (r in 1:rows) {
+    for (c in 1:col) {
+      index <-  index + 1
+      if (tbc != "" && index == tbcIndex) {
+        plot <- ggplot2::ggplot() + ggplot2::theme_void() +
+          ggplot2::annotate(geom = "text", x = 0, y = 0, label = tbc, size = 10)
+      } else if(index == emptyIndex) {
+        plot <- ggplot2::ggplot() + ggplot2::theme_void()
+      } else {
+        plot <- plotList[[index]]
+      }
+      plotMat[[r, c]] <- plot
+    }
+  }
+  return(plotMat)
+}
+
+.getFromToSampleShow <- function(type = c("first", "last", "range", "all"), maxSamples, singleValue = "", start ="", stop = ""){
+  if (type == "first") {
```

Comment:
for safety, perhaps add `type <- match.arg(type)`?
```suggestion
.getFromToSampleShow <- function(type = c("first", "last", "range", "all"), maxSamples, singleValue = "", start ="", stop = ""){
  type <- match.arg(type)
  if (type == "first") {
```